ember/app/ui/cases/detail/index/template.hbs

```diff
@@ -71,19 +71,27 @@
         {{t "documents.edit"}}
       </LinkTo>
     
-      <h3>{{t "documents.submit"}}</h3>
+      <h3 {{did-insert (perform this.computeDocument)}}>{{t "documents.submit"}}</h3>
       {{#if this.submitDisabled}}
         <p>{{t "documents.noSubmitText"}}</p>
       {{else}}
         <p>{{t "documents.submitText"}}</p> 
       {{/if}}
-      <WorkItemButton
-        @mutation="complete"
-        @workItemId={{@model.workItems.edges.firstObject.node.id}}
-        @label={{t "documents.submit"}}
-        @disabled={{this.submitDisabled}}
-        @onSuccess={{this.transitionToCase}}
-      />
+      {{#if this.computeDocument.isRunning}}
+        <div class="uk-text-center uk-padding">
+          <UkSpinner @ratio={{2}} />
+        </div>
+      {{else}}
+        <DocumentValidity @document={{this.computeDocument.lastSuccessful.value}} as |isValid|>
+          <WorkItemButton
+            @mutation="complete"
+            @workItemId={{@model.workItems.edges.firstObject.node.id}}
+            @label={{t "documents.submit"}}
+            @disabled={{and this.submitDisabled (not isValid)}}
+            @onSuccess={{this.transitionToCase}}
+          />
+        </DocumentValidity>
```

Comment:
As discussed: Let's move the submit button to the document tab instead.

Reply:
added override component